[Sync] `Result::unwrap()` on an `Err` value: PoisonError { .. } -> Panic at anki/rslib/src/backend/sync.rs:249: called `Option::unwrap()` on a `None` value

https://ankidroid.org/acra/app/1/bug/252979/report/7a438d99-acee-4da5-be87-798e0cee6f29

```
net.ankiweb.rsdroid.BackendException$BackendDbException$BackendDbFileTooOldException: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
	at net.ankiweb.rsdroid.BackendException$BackendFatalError.<init>(BackendException.java:99)
	at net.ankiweb.rsdroid.BackendException$Companion.fromError(BackendException.kt:108)
	at net.ankiweb.rsdroid.BackendKt.unpackResult(Backend.kt:281)
	at net.ankiweb.rsdroid.BackendKt.access$unpackResult(Backend.kt:1)
	at net.ankiweb.rsdroid.Backend.runMethodRaw$lambda$1(Backend.kt:118)
	at net.ankiweb.rsdroid.Backend.withBackend(Backend.kt:131)
	at net.ankiweb.rsdroid.Backend.runMethodRaw(Backend.kt:117)
	at anki.backend.GeneratedBackend.openCollectionRaw(GeneratedBackend.kt:112)
	at anki.backend.GeneratedBackend.openCollection(GeneratedBackend.kt:119)
	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:98)
	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:57)
	at com.ichi2.libanki.Storage.openDB$AnkiDroid_playRelease(Storage.kt:52)
	at com.ichi2.libanki.Collection.reopen(Collection.kt:213)
	at com.ichi2.libanki.Collection.reopen$default(Collection.kt:210)
	at com.ichi2.libanki.Collection.<init>(Collection.kt:145)
	at com.ichi2.libanki.Storage.collection(Storage.kt:40)
	at com.ichi2.anki.CollectionManager.ensureOpenInner(CollectionManager.kt:245)
	at com.ichi2.anki.CollectionManager.withCol$lambda$1(CollectionManager.kt:118)
	at com.ichi2.anki.CollectionManager$withQueue$3.invokeSuspend(CollectionManager.kt:103)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.java:111)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)

```

```
--------- beginning of main
11-06 11:52:51.195 I/AnkiDroid(13131): DeckPicker::x::onPause
11-06 11:52:51.196 I/AnkiDroid(13131): DeckPicker::x::onStop
11-06 11:52:51.228 I/AnkiDroid(13131): DeckPicker::x::onViewDestroyed
11-06 11:52:51.228 I/AnkiDroid(13131): DeckPicker::x::onAttach
11-06 11:52:51.229 I/AnkiDroid(13131): DeckPicker::x::onCreate
11-06 11:52:51.236 W/AnkiDroid(13131): x/ java.io.IOException: Cannot run program "sqlite3": error=2, No such file or directory
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1050)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:712)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:536)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:433)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog(DatabaseErrorDialog.kt:91)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.DialogFragment.prepareDialog(DialogFragment.java:925)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:839)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1760)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:547)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:286)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2211)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2106)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2049)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.executePendingTransactions(FragmentManager.java:779)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity$Companion.showDialogFragment(AnkiActivity.java:702)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity$Companion.showDialogFragment(AnkiActivity.java:687)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showDialogFragment$lambda$4(AnkiActivity.kt:462)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.app.Activity.runOnUiThread(Activity.java:7865)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showDialogFragment(AnkiActivity.kt:461)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.kt:489)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.DeckPicker.onCreate$lambda$13(DeckPicker.kt:473)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.kt:473)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.DeckPicker.showDatabaseErrorDialog(DeckPicker.kt:1697)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog$lambda$10$lambda$9(DatabaseErrorDialog.kt:196)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.utils.AlertDialogFacadeKt.listItems$lambda$13(AlertDialogFacade.kt:335)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.appcompat.app.AlertController$AlertParams$3.onItemClick(AlertController.java:1068)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AdapterView.performItemClick(AdapterView.java:376)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView.performItemClick(AbsListView.java:1362)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3658)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView$3.run(AbsListView.java:4844)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Handler.handleCallback(Handler.java:958)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Handler.dispatchMessage(Handler.java:99)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Looper.loopOnce(Looper.java:230)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Looper.loop(Looper.java:319)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.app.ActivityThread.main(ActivityThread.java:8919)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.reflect.Method.invoke(Native Method)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
11-06 11:52:51.236 W/AnkiDroid(13131): Caused by: java.io.IOException: error=2, No such file or directory
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.UNIXProcess.forkAndExec(Native Method)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.UNIXProcess.<init>(UNIXProcess.java:133)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessImpl.start(ProcessImpl.java:141)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
11-06 11:52:51.236 W/AnkiDroid(13131): 	... 32 more
11-06 11:52:51.236 W/AnkiDroid(13131): 
11-06 11:52:51.236 W/AnkiDroid(13131): java.io.IOException: Cannot run program "sqlite3": error=2, No such file or directory
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1050)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:712)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:536)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.Runtime.exec(Runtime.java:433)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog(DatabaseErrorDialog.kt:91)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.DialogFragment.prepareDialog(DialogFragment.java:925)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:839)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1760)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:547)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:286)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2211)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2106)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2049)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.fragment.app.FragmentManager.executePendingTransactions(FragmentManager.java:779)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity$Companion.showDialogFragment(AnkiActivity.java:702)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity$Companion.showDialogFragment(AnkiActivity.java:687)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showDialogFragment$lambda$4(AnkiActivity.kt:462)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.app.Activity.runOnUiThread(Activity.java:7865)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showDialogFragment(AnkiActivity.kt:461)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.kt:489)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.DeckPicker.onCreate$lambda$13(DeckPicker.kt:473)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.kt:473)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.DeckPicker.showDatabaseErrorDialog(DeckPicker.kt:1697)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog$lambda$10$lambda$9(DatabaseErrorDialog.kt:196)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.ichi2.utils.AlertDialogFacadeKt.listItems$lambda$13(AlertDialogFacade.kt:335)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at androidx.appcompat.app.AlertController$AlertParams$3.onItemClick(AlertController.java:1068)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AdapterView.performItemClick(AdapterView.java:376)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView.performItemClick(AbsListView.java:1362)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3658)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.widget.AbsListView$3.run(AbsListView.java:4844)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Handler.handleCallback(Handler.java:958)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Handler.dispatchMessage(Handler.java:99)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Looper.loopOnce(Looper.java:230)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.os.Looper.loop(Looper.java:319)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at android.app.ActivityThread.main(ActivityThread.java:8919)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.reflect.Method.invoke(Native Method)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
11-06 11:52:51.236 W/AnkiDroid(13131): Caused by: java.io.IOException: error=2, No such file or directory
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.UNIXProcess.forkAndExec(Native Method)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.UNIXProcess.<init>(UNIXProcess.java:133)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessImpl.start(ProcessImpl.java:141)
11-06 11:52:51.236 W/AnkiDroid(13131): 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
11-06 11:52:51.236 W/AnkiDroid(13131): 	... 32 more
11-06 11:52:51.275 I/AnkiDroid(13131): DeckPicker::x::onStart
11-06 11:52:51.275 I/AnkiDroid(13131): DeckPicker::x::onResume
11-06 11:52:53.538 I/AnkiDroid(13131): (Re)opening Database: /storage/emulated/0/Android/data/com.ichi2.anki/files/AnkiDroid/collection.anki2
11-06 11:52:53.554 I/AnkiDroid(13131): DeckPicker::onPause
11-06 11:52:53.554 I/AnkiDroid(13131): DeckPicker::x::onPause
11-06 11:52:53.554 I/AnkiDroid(13131): DeckPicker::x::onStop
11-06 11:52:53.554 I/AnkiDroid(13131): DeckPicker::x::onViewDestroyed
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDestroy
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDetach
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDestroy
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDetach
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDestroy
11-06 11:52:53.555 I/AnkiDroid(13131): DeckPicker::x::onDetach
11-06 11:52:53.565 E/AnkiDroid(13131): F3/ q7.c: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendException$BackendFatalError.<init>(BackendException.java:99)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendException$Companion.fromError(BackendException.kt:108)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendKt.unpackResult(Backend.kt:281)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendKt.access$unpackResult(Backend.kt:1)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.runMethodRaw$lambda$1(Backend.kt:118)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.withBackend(Backend.kt:131)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.runMethodRaw(Backend.kt:117)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at anki.backend.GeneratedBackend.openCollectionRaw(GeneratedBackend.kt:112)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at anki.backend.GeneratedBackend.openCollection(GeneratedBackend.kt:119)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:98)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:57)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Storage.openDB$AnkiDroid_playRelease(Storage.kt:52)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.reopen(Collection.kt:213)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.reopen$default(Collection.kt:210)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.<init>(Collection.kt:145)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Storage.collection(Storage.kt:40)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager.ensureOpenInner(CollectionManager.kt:245)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager.withCol$lambda$1(CollectionManager.kt:118)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager$withQueue$3.invokeSuspend(CollectionManager.kt:103)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.java:111)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)
11-06 11:52:53.565 E/AnkiDroid(13131): 
11-06 11:52:53.565 E/AnkiDroid(13131): net.ankiweb.rsdroid.BackendException$BackendDbException$BackendDbFileTooOldException: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendException$BackendFatalError.<init>(BackendException.java:99)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendException$Companion.fromError(BackendException.kt:108)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendKt.unpackResult(Backend.kt:281)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.BackendKt.access$unpackResult(Backend.kt:1)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.runMethodRaw$lambda$1(Backend.kt:118)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.withBackend(Backend.kt:131)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.runMethodRaw(Backend.kt:117)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at anki.backend.GeneratedBackend.openCollectionRaw(GeneratedBackend.kt:112)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at anki.backend.GeneratedBackend.openCollection(GeneratedBackend.kt:119)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:98)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at net.ankiweb.rsdroid.Backend.openCollection(Backend.kt:57)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Storage.openDB$AnkiDroid_playRelease(Storage.kt:52)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.reopen(Collection.kt:213)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.reopen$default(Collection.kt:210)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Collection.<init>(Collection.kt:145)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.libanki.Storage.collection(Storage.kt:40)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager.ensureOpenInner(CollectionManager.kt:245)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager.withCol$lambda$1(CollectionManager.kt:118)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at com.ichi2.anki.CollectionManager$withQueue$3.invokeSuspend(CollectionManager.kt:103)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.java:111)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
11-06 11:52:53.565 E/AnkiDroid(13131): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)
11-06 11:52:53.685 I/AnkiDroid(13131): notification permission: false
11-06 11:52:53.685 I/AnkiDroid(13131): DeckPicker::onResume
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
@@ -34,6 +34,7 @@ import com.ichi2.anki.dialogs.SyncErrorDialog
 import com.ichi2.anki.preferences.sharedPrefs
 import com.ichi2.anki.settings.Prefs
 import com.ichi2.anki.snackbar.showSnackbar
+import com.ichi2.anki.ui.internationalization.toSentenceCase
 import com.ichi2.anki.worker.SyncMediaWorker
 import com.ichi2.libanki.ChangeManager.notifySubscribersAllValuesChanged
 import com.ichi2.libanki.createBackup
@@ -385,7 +386,7 @@ suspend fun monitorMediaSync(deckPicker: DeckPicker) {
         withContext(Dispatchers.Main) {
             AlertDialog
                 .Builder(deckPicker)
-                .setTitle(TR.syncMediaLogTitle())
+                .setTitle(TR.syncMediaLogTitle().toSentenceCase(deckPicker, R.string.sentence_sync_media_log))
                 .setMessage("")
                 .setPositiveButton(R.string.dialog_continue) { _, _ ->
                     scope.cancel()

--- a/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
@@ -42,6 +42,7 @@ import com.ichi2.libanki.syncCollection
 import com.ichi2.libanki.syncLogin
 import com.ichi2.preferences.VersatileTextWithASwitchPreference
 import com.ichi2.utils.NetworkUtils
+import com.ichi2.utils.dismissSafely
 import kotlinx.coroutines.CancellationException
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
@@ -416,7 +417,7 @@ suspend fun monitorMediaSync(deckPicker: DeckPicker) {
         } catch (_: Exception) {
             showMessage(TR.syncMediaFailed())
         } finally {
-            dialog.dismiss()
+            dialog.dismissSafely()
         }
     }
 }

--- a/AnkiDroid/src/main/java/com/ichi2/anki/worker/SyncMediaWorker.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/worker/SyncMediaWorker.kt
@@ -71,8 +71,10 @@ class SyncMediaWorker(
                     }
                 }
 
+            // The collection must be open, but we should not block collection operations while
+            // `syncMedia` is executing, the app should be usable during a background media sync
+            CollectionManager.getColUnsafe().backend.syncMedia(auth)
             val backend = CollectionManager.getBackend()
-            backend.syncMedia(auth)
 
             delay(1000) // avoid notifications if sync occurs too quickly
             if (backend.mediaSyncStatus().active) {

--- a/AnkiDroid/src/main/java/com/ichi2/utils/AlertDialogFacade.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/utils/AlertDialogFacade.kt
@@ -44,6 +44,7 @@ import com.google.android.material.textfield.TextInputLayout
 import com.ichi2.anki.R
 import com.ichi2.themes.Themes
 import com.ichi2.ui.FixedTextView
+import com.ichi2.utils.HandlerUtils.executeOnMainThread
 import timber.log.Timber
 
 /** Wraps [DialogInterface.OnClickListener] as we don't need the `which` parameter */
@@ -451,5 +452,22 @@ fun AlertDialog.Builder.titleWithHelpIcon(
     customTitleView.findViewById<ImageView>(R.id.help_icon).setOnClickListener { v ->
         Timber.i("dialog help icon click")
         block.onClick(v)
+    }
+}
+
+/** Calls [AlertDialog.dismiss], ignoring errors */
+fun AlertDialog.dismissSafely() {
+    // The exception will be uncaught if not run on the main thread.
+    executeOnMainThread {
+        try {
+            // safer to catch the exception to be sure dismiss() was called
+            dismiss()
+        } catch (e: IllegalArgumentException) {
+            if (window == null || !isShowing) {
+                Timber.d(e, "Dialog not attached to window manager")
+                return@executeOnMainThread
+            }
+            Timber.w(e, "Dialog not attached to window manager")
+        }
     }
 }

--- a/AnkiDroid/src/main/java/com/ichi2/utils/HandlerUtils.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/utils/HandlerUtils.kt
@@ -88,4 +88,18 @@ object HandlerUtils {
             time,
         )
     }
+
+    /**
+     * Executes a method on the main thread. Either immediately, or via
+     * [HandlerUtils.executeFunctionUsingHandler]
+     *
+     * @param function The function which needs to be executed.
+     */
+    fun executeOnMainThread(function: () -> Unit) {
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            function()
+        } else {
+            executeFunctionUsingHandler { function() }
+        }
+    }
 }

--- a/AnkiDroid/src/main/res/values/sentence-case.xml
+++ b/AnkiDroid/src/main/res/values/sentence-case.xml
@@ -48,5 +48,6 @@ undoActionUndone()
     <string name="sentence_check_media_delete_unused">Delete unused</string>
     <string name="sentence_choose_tags">Choose tags</string>
     <string name="sentence_selective_study">Selective study</string>
+    <string name="sentence_sync_media_log">Media sync log</string>
 
 </resources>

--- a/AnkiDroid/src/test/java/com/ichi2/anki/ui/internationalization/SentenceCaseTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/ui/internationalization/SentenceCaseTest.kt
@@ -39,6 +39,12 @@ class SentenceCaseTest : RobolectricTest() {
             assertThat(TR.actionsSetDueDate().toSentenceCase(this, R.string.sentence_set_due_date), equalTo("Set due date"))
             assertThat(TR.actionsCustomStudy().toSentenceCase(this, R.string.sentence_custom_study), equalTo("Custom study"))
             assertThat(TR.emptyCardsWindowTitle().toSentenceCase(this, R.string.sentence_empty_cards), equalTo("Empty cards"))
+            assertThat("syncMediaLogTitle", TR.syncMediaLogTitle(), equalTo("Media Sync Log"))
+            assertThat(
+                "sentence_sync_media_log",
+                TR.syncMediaLogTitle().toSentenceCase(this, R.string.sentence_sync_media_log),
+                equalTo("Media sync log"),
+            )
 
             assertThat("Toggle Suspend".toSentenceCase(this, R.string.sentence_toggle_suspend), equalTo("Toggle suspend"))
             assertThat("Ook? Ook?".toSentenceCase(this, R.string.sentence_toggle_suspend), equalTo("Ook? Ook?"))

